src/Bicep.Cli/Commands/BuildParamsCommand.cs

```diff
@@ -58,83 +58,56 @@ public async Task<int> RunAsync(BuildParamsArguments args)
                 logger.LogWarning(CliResources.ResourceTypesDisclaimerMessage);
             }
 
-            if(bicepInputPath != null && !IsBicepFile(bicepInputPath))
+            if(bicepFileArgPath != null && !IsBicepFile(bicepFileArgPath))
             {
-                throw new CommandLineException($"{bicepInputPath} is not a bicep file");
+                throw new CommandLineException($"{bicepFileArgPath} is not a bicep file");
             }
 
-            if(features.ParamsFilesEnabled && IsBicepparamsFile(paramsInputPath)) 
+            if(!features.ParamsFilesEnabled) 
             {
-                var paramsCompilation = await compilationService.CompileAsync(paramsInputPath, args.NoRestore);
+                logger.LogError(CliResources.UnableToCompileParamsFile, paramsInputPath, nameof(ExperimentalFeaturesEnabled.ParamsFiles));
+                return 1;
+            }
 
-                var bicepSemanticModel = GetBicepSemanticModelFromParamsCompilation(paramsCompilation);
+            if (!IsBicepparamsFile(paramsInputPath))
+            {
+                logger.LogError(CliResources.UnrecognizedFileExtensionMessage, paramsInputPath);
+                return 1;
+            }
 
-                if(bicepSemanticModel != null && bicepInputPath != null) 
-                {   
-                    var bicepFilePath = bicepSemanticModel.Root.FileUri;
-                    
-                    if(!bicepFilePath.Equals(PathHelper.FilePathToFileUrl(bicepInputPath)))
-                    {
-                        throw new CommandLineException($"Bicep file {bicepInputPath} provided with --bicep-file option doesn't match the Bicep file {bicepSemanticModel.Root.Name} referenced by the using declaration in the parameters file");
-                    }
+            var paramsCompilation = await compilationService.CompileAsync(paramsInputPath, args.NoRestore);
+
+            var paramsSemanticModel = paramsCompilation.GetEntrypointSemanticModel();
+
+            //Failure scenario is ignored since a diagnostic for it would be emitted during semantic analysis 
+            if(paramsSemanticModel.Root.TryGetBicepFileSemanticModelViaUsing(out var bicepSemanticModel, out _))
+            {
+                var bicepFileUsingPathUri = bicepSemanticModel.Root.FileUri;
+
+                if(bicepFileArgPath is {} && !bicepFileUsingPathUri.Equals(PathHelper.FilePathToFileUrl(bicepFileArgPath))) 
+                {                   
+                    throw new CommandLineException($"Bicep file {bicepFileArgPath} provided with --bicep-file option doesn't match the Bicep file {bicepSemanticModel.Root.Name} referenced by the using declaration in the parameters file");
                 }
```

Comment:
Should we just drop the `--bicep-file` argument altogether? With the changes from this PR, the params file must have a `using` statement, and the `--bicep-file` arg (if supplied) must match the `using` statement, so `--bicep-file` is not supplying any additional information and is just creating a new opportunity for error.

Reply:
In az cli/az pwsh, the user will always provide a bicep file to deploy.  The main reason we need `--bicep-file` is to take the bicep file reference that user provided (in az cli/az pwsh) and match it with the `using` declaration. In case those two differ, we throw some error so the user doesn't accidentally deploy the wrong file (because the template that gets complied/deployed is the one in `using` declaration)

Other than az cli/az pwsh reference check use case, the `--bicep-file` arg is auxiliary and not really needed if the user is only trying to compile a param file through `params-build` command.  This is just a short term solution for the issue and might try to create only param file based deployment based on user requests in future.